src/freenet/io/xfer/BulkTransmitter.java

```diff
@@ -247,7 +247,7 @@ public boolean send() throws DisconnectedException {
 		long lastSentPacket = System.currentTimeMillis();
 outer:	while(true) {
 			int max = Math.min(Integer.MAX_VALUE, prb.blocks);
-			max = Math.min(max, (int)Math.min(Integer.MAX_VALUE, peer.getThrottleWindowSize()));
+			max = Math.min(max, Math.min(Integer.MAX_VALUE, peer.getThrottleWindowSize()));
```

Comment:
If `peer.getThrottleWindowSize()` returns an `int`, the whole `Math.min` expression can go.